boolean_algebra/karnaugh_map_simplification.py

```diff
@@ -0,0 +1,44 @@
+# https://www.allaboutcircuits.com/technical-articles/karnaugh-map-boolean-algebraic-simplification-technique/
+
+
+def simplify_kmap(kmap: list[list[int]]) -> str:
+    """
+    Simplify the K-Map.
+
+    >>> kmap = [[0, 1], [1, 1]]
+    >>> simplify_kmap(kmap)
+    "A'B + AB' + AB"
```

Comment:
Let's add some tests that break our algorithm.  Tests should show how the code work but also how it breaks.
```suggestion
    >>> simplify_kmap(kmap=[[0, 1], [1, 1]])
    "A'B + AB' + AB"
    >>> simplify_kmap(kmap=[[0, 0], [0, 0]])
    "A'B + AB' + AB"
    >>> simplify_kmap(kmap=[[0, 1], [1, -1]])
    "A'B + AB' + AB"
    >>> simplify_kmap(kmap=[[0, 1], [1, 2]])
    "A'B + AB' + AB"
    >>> simplify_kmap(kmap=[[0, 1], [1, 1.1]])
    "A'B + AB' + AB"
    >>> simplify_kmap(kmap=[[0, 1], [1, 'a']])
    "A'B + AB' + AB"
```